From 2f3de7c878438b52fce7d0213dde5f66f68fb370 Mon Sep 17 00:00:00 2001
From: "Chung, Kevin" <kevin.chung@intel.com>
Date: Tue, 10 Feb 2015 17:24:47 +0800
Subject: [PATCH] Merge the patches from product variant cht_rvp to
 cht_cr_rvp. Signed-off-by: Chung, Kevin
 <kevin.chung@intel.com>

Change-Id: I5e86c9ab1de2b42dde0fe12cd677fb7d42e3eb67
---
 cht_cr_rvp/BoardConfig.mk |   16 +++++++-------
 cht_cr_rvp/device.mk      |   24 ++++++++++++++------
 cht_cr_rvp/init.rc        |   53 +++++++++++++++++++++++++++++----------------
 cht_cr_rvp/mixins.spec    |    2 +-
 4 files changed, 60 insertions(+), 35 deletions(-)

diff --git device/intel/cherrytrail/cht_cr_rvp/BoardConfig.mk device/intel/cherrytrail/cht_cr_rvp/BoardConfig.mk
index 0da9927..0a522aa 100755
--- device/intel/cherrytrail/cht_cr_rvp/BoardConfig.mk
+++ device/intel/cherrytrail/cht_cr_rvp/BoardConfig.mk
@@ -258,18 +258,18 @@ BOARD_BLUETOOTH_BDROID_BUILDCFG_INCLUDE_DIR := device/intel/common/bluetooth/def
 DEVICE_PACKAGE_OVERLAYS += device/intel/common/bluetooth/overlay-bt-pan
 
 ##############################################################
-# Source: device/intel/mixins/groups/wlan/rtl8723bs/BoardConfig.mk
+# Source: device/intel/mixins/groups/wlan/iwlwifi/BoardConfig.mk
 ##############################################################
-ADDITIONAL_DEFAULT_PROPERTIES += wifi.interface=wlan0
-WPA_SUPPLICANT_VERSION := VER_2_1_DEVEL
+# This enables the wpa wireless driver
+BOARD_HOSTAPD_DRIVER := NL80211
 BOARD_WPA_SUPPLICANT_DRIVER := NL80211
-BOARD_WLAN_DEVICE := rtl
-BOARD_WPA_SUPPLICANT_PRIVATE_LIB := lib_driver_cmd_rtl
+WPA_SUPPLICANT_VERSION := VER_2_1_DEVEL_WCS
 
-DEVICE_PACKAGE_OVERLAYS += device/intel/common/wlan/overlay-wifi-display
+# Enabling iwlwifi
+BOARD_USING_INTEL_IWL := true
 
-COMBO_CHIP_VENDOR := rtl
-COMBO_CHIP := rtl8723
+COMBO_CHIP_VENDOR := intel
+COMBO_CHIP := iwlwifi
 ##############################################################
 # Source: device/intel/mixins/groups/widi/gen/BoardConfig.mk
 ##############################################################
diff --git device/intel/cherrytrail/cht_cr_rvp/device.mk device/intel/cherrytrail/cht_cr_rvp/device.mk
index 9d30ff4..73f334a 100644
--- device/intel/cherrytrail/cht_cr_rvp/device.mk
+++ device/intel/cherrytrail/cht_cr_rvp/device.mk
@@ -271,16 +271,26 @@ PRODUCT_PACKAGES += \
     Tag
 endif
 ##############################################################
-# Source: device/intel/mixins/groups/wlan/rtl8723bs/product.mk
+# Source: device/intel/mixins/groups/wlan/iwlwifi/product.mk
 ##############################################################
-# Uses same supplicant config as the bcm43241
+PRODUCT_PACKAGES += \
+    hostapd \
+    hostap_cli \
+    wpa_supplicant \
+    wpa_cli
+
+#copy iwlwifi wpa p2p config files
+ PRODUCT_COPY_FILES += \
+        device/intel/common/wlan/iwlwifi/p2p_supplicant.conf:system/etc/wifi/p2p_supplicant.conf \
+        device/intel/common/wlan/iwlwifi/wpa_supplicant.conf:system/etc/wifi/wpa_supplicant.conf \
+        frameworks/native/data/etc/android.hardware.wifi.xml:system/etc/permissions/android.hardware.wifi.xml \
+        frameworks/native/data/etc/android.hardware.wifi.direct.xml:system/etc/permissions/android.hardware.wifi.direct.xml
+
 PRODUCT_COPY_FILES += \
-    device/intel/common/wlan/wpa_supplicant-common.conf:system/etc/wifi/wpa_supplicant.conf \
-    device/intel/common/wlan/wpa_supplicant_overlay.conf:system/etc/wifi/wpa_supplicant_overlay.conf \
-    frameworks/native/data/etc/android.hardware.wifi.xml:system/etc/permissions/android.hardware.wifi.xml \
-    frameworks/native/data/etc/android.hardware.wifi.direct.xml:system/etc/permissions/android.hardware.wifi.direct.xml
+        device/intel/common/wlan/iwlwifi/load_iwlwifi.sh:system/bin/load_iwlwifi.sh
 
-PRODUCT_PACKAGES += wpa_supplicant
+$(call inherit-product-if-exists, vendor/intel/fw/iwl/iwl-fw-8000-cht-gmin.mk)
+$(call inherit-product-if-exists, kernel/gmin/uefi/modules/wlan/iwlwifi/Androidiwlwifi.mk)
 
 ##############################################################
 # Source: device/intel/mixins/groups/widi/gen/product.mk
diff --git device/intel/cherrytrail/cht_cr_rvp/init.rc device/intel/cherrytrail/cht_cr_rvp/init.rc
index b113a77..49851d5 100644
--- device/intel/cherrytrail/cht_cr_rvp/init.rc
+++ device/intel/cherrytrail/cht_cr_rvp/init.rc
@@ -268,43 +268,58 @@ on post-fs-data
     # Create data folder for NFC
     mkdir /data/nfc 0770 nfc nfc
 ##############################################################
-# Source: device/intel/mixins/groups/wlan/rtl8723bs/init.rc
+# Source: device/intel/mixins/groups/wlan/iwlwifi/init.rc
 ##############################################################
-service p2p_supplicant /system/bin/wpa_supplicant \
-    -ip2p0 -Dnl80211 -c/data/misc/wifi/p2p_supplicant.conf -N \
-    -pforce_single_radio=1 \
-    -iwlan0 -Dnl80211 -c/data/misc/wifi/wpa_supplicant.conf \
-    -I/system/etc/wifi/wpa_supplicant_overlay.conf \
-    -O/data/misc/wifi/sockets -puse_p2p_group_interface=1,force_single_radio=1 \
-    -e/data/misc/wifi/entropy.bin -g@android:wpa_wlan0
+on post-fs-data
+    chmod 0660 /data/misc/wifi/p2p_supplicant.conf
+    # create config WiFi NVM folder
+    mkdir /config/wlan
+    chmod 755 /config/wlan
+    start wlan_rest_nvm
+    wait /config/wlan/iwl_nvm.bin
+    start load_iwlwifi
+
+service wpa_supplicant /system/bin/wpa_supplicant -Dnl80211 -iwlan0 -g@android:wpa_wlan0 -c/data/misc/wifi/wpa_supplicant.conf -dt \
+    -e /data/misc/wifi/entropy.bin
     class main
     socket wpa_wlan0 dgram 660 wifi wifi
     disabled
     oneshot
 
-service dhcpcd_wlan0 /system/bin/dhcpcd -aABDKL
-     class main
-     disabled
-     oneshot
+service dhcpcd_wlan0 /system/bin/dhcpcd -ABDKL
+    class main
+    disabled
+    oneshot
 
 service iprenew_wlan0 /system/bin/dhcpcd -n
-     class main
-     disabled
-     oneshot
+    class main
+    disabled
+    oneshot
 
-service dhcpcd_p2p /system/bin/dhcpcd -aABKL
+service p2p_supplicant /system/bin/wpa_supplicant \
+    -dt -i wlan0 -Dnl80211 -c/data/misc/wifi/wpa_supplicant.conf -g@android:wpa_wlan0 \
+    -m/system/etc/wifi/p2p_supplicant.conf \
+    -O/data/misc/wifi/sockets -e /data/misc/wifi/entropy.bin
     class main
+    socket wpa_wlan0 dgram 660 wifi wifi
+    disabled
+    oneshot
+
+service dhcpcd_p2p /system/bin/dhcpcd -aABKL
     disabled
     oneshot
 
 service iprenew_p2p /system/bin/dhcpcd -n
+    disabled
+    oneshot
+
+service load_iwlwifi /system/bin/sh /system/bin/load_iwlwifi.sh
     class main
+    user root
+    group system wifi
     disabled
     oneshot
 
-on boot
-    insmod /system/lib/modules/cfg80211.ko
-    insmod /system/lib/modules/8723bs.ko
 ##############################################################
 # Source: device/intel/mixins/groups/widi/gen/init.rc
 ##############################################################
diff --git device/intel/cherrytrail/cht_cr_rvp/mixins.spec device/intel/cherrytrail/cht_cr_rvp/mixins.spec
index f915709..5ae6564 100644
--- device/intel/cherrytrail/cht_cr_rvp/mixins.spec
+++ device/intel/cherrytrail/cht_cr_rvp/mixins.spec
@@ -20,7 +20,7 @@ camera: isp3
 rfkill: true(force_disable=gps bluetooth)
 bluetooth: rtl8723bs
 nfc: pn547
-wlan: rtl8723bs
+wlan: iwlwifi
 widi: gen
 audio: cht-tlv320aic31xx
 media: ufo
-- 
1.7.9.5

